docs/architecture/incremental.md

```diff
@@ -0,0 +1,23 @@
+# Configuration: Streams & Incremental Syncs
+
+## AirbyteStream
+* The `AirbyteStream` has two sections:
```

Comment:
this is a relatively sweeping change in that it is changing the structure of existing fields. I _think_ it's worth it so that we have more clarity on what each of these things are for. But I'm open to the argument that we should prefer skipping the breaking change.